src/py/flwr/server/superlink/state/state.py

```diff
@@ -150,9 +150,26 @@ def get_nodes(self, run_id: int) -> Set[int]:
         """
 
     @abc.abstractmethod
-    def create_run(self) -> int:
+    def create_run(self, fab_id: str, fab_version: str) -> int:
         """Create one run."""
 
+    @abc.abstractmethod
+    def get_run(self, run_id: int) -> Tuple[str, str]:
+        """Retrieve information about the run with the specified `run_id`.
+
+        Parameters
+        ----------
+        run_id : int
+            The identifier of the run.
+
+        Returns
+        -------
+        Tuple[str, str]
+            A tuple containing two strings:
+            - `fab_id`: The identifier of the Flower Fabric used in the specified run.
+            - `fab_version`: The version of the Flower Fabric used in the specified run.
```

Comment:
Let's add the `run_id` here as well. We can use a tuple for now, but we should probably switch to a dataclass `Run` in the future.

Reply:
I am okay with adding `run_id` here, but just a little bit confused. To me, it seems like `get_run` is the method that retrieving the details of a run given its ID, so naturally the definition of the method would be ID -> details. May I know why we also return run_id?

Reply:
You're right, we could equally well not do this. There's no strong reason for doing this. It's more of a convenience thing. When the `Run` object gets more complex in the future, we can just return the full object and then serialize it and send it back to the SuperNode.